src/lib/core/CHIPTLVReader.cpp

```diff
@@ -255,6 +255,17 @@ CHIP_ERROR TLVReader::Get(double & v)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR TLVReader::Get(chip::ByteSpan & v)
+{
+    const uint8_t * val;
```

Comment:
Shouldn't this test somewhere that we're an octet string?  The documentation sounded like this should only work for octet string, but the code works for char string too, getting it out as a bytespan....

Reply:
This is correct. Unless #7963 lands `CHAR_STRING` are still reads as `ByteSpan`. For #7963 I will add a `TLVReader::Get(chip::Span<char> & v);` method and I will add checks for `TLVTypeIsByteString` here and `TLVTypeIsUTF8String` in the new method.